routes/tests/shopifyAuth.test.js

```diff
@@ -0,0 +1,58 @@
+const request = require('supertest');
+const http = require('http');
+const express = require('express');
+
+const {MemoryStrategy} = require('../../strategies');
+const createShopifyAuthRouter = require('../shopifyAuth');
+
+let server;
+describe('shopifyAuth', async () => {
+  afterEach(() => {
+    server.close();
+  });
+
+  describe('/', () => {
+    it('responds to get requests by returning a redirect page', async () => {
+      const {app, server} = await createServer();
+      const {status, text} = await request(app).get('/?shop=shop1').then((resp) => resp);
+
+      expect(status).toBe(200);
+      expect(text).toMatchSnapshot();
+
+      server.close();
+    });
+
+    it('responds with a 400 when no shop query parameter is given', async () => {
+      const {app, server} = await createServer();
+      const {status, text} = await request(app).get('/').then((resp) => resp);
+
+      expect(status).toBe(400);
+      expect(text).toMatchSnapshot();
+
+      server.close();
+    });
+  });
+})
+
+function createServer({afterAuth = jest.fn()} = {}) {
```

Comment:
Why choose `jest.fn()` here? Would that also be an option for our thing to invoke `then` for `supertest`?

Reply:
`jest.fn()` is a jest mock, so we're returning it here so we can do things like test that it was called with what we expect at the end of the function.